### PR TITLE
fix: substitute UT_ASSERTs with asserts from GTEST part 4

### PR DESCRIPTION
**Pull request by EuphoricThinking**
The PR is a part of a major fix, which will be provided in multiple files.

Ref. #569 

canQuery*(...) are changed to void. The success, failure or the need to skip the tests are determined in the function body using GTEST asserts, which set the appropriate flags for the current test. The flags are checked then in the fixture. The solution is inspired by [GTEST docs.](https://google.github.io/googletest/advanced.html#checking-for-failures-in-the-current-test).

The condition using ::memspaceGetTest::IsSkipped() in memspaceProviderTest is replaced due to its ineffectiveness: it has used the method from superclass for itself, before skipping any tests.

EXPECT_* and ADD_FAILURE() interefere with GTEST_SKIP() when used in a non-void helper function - tests are run in spite of labelling as ,,skipped" by the fixture.

### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

File: test/memspaces/memspace_highest_bandwidth.cpp [hunk 1]
```diff
@@ -12,13 +12,23 @@
 static bool canQueryBandwidth(size_t nodeId) {
     hwloc_topology_t topology = nullptr;
     int ret = hwloc_topology_init(&topology);
-    UT_ASSERTeq(ret, 0);
+
+    if (!GTEST_OUT_EQ(ret, 0)) {
```
Comment (PatKamin) [inline test/memspaces/memspace_highest_bandwidth.cpp:16]: Can the `EXPECT_EQ` macro be used instead? What is the advantage of using the `GTEST_OUT_EQ` macro over `EXPECT_EQ`?

Comment (EuphoricThinking) [reply test/memspaces/memspace_highest_bandwidth.cpp:16]: I have tried EXPECT_EQ() and it interferes with GTEST_SKIP(): although the tests are labeled as ,,skipped" in the fixture, they are executed. For example, failed ASSERT_NE(sth, nullptr) are outputted, because sth has been never initialized since the fixture stopped the SetUp() execution after GTEST_SKIP(). 

 ADD_FAILURE() gives similar result to EXPECT_EQ() (tests are eventually run).

Therefore I have decided to display the message and to manipulate the program execution only in the given fixture.

Comment (PatKamin) [reply test/memspaces/memspace_highest_bandwidth.cpp:16]: If we need these GTEST checks, then I'm ok with that. Please, consider if we could get rid of the GTEST macros in this function and ie. return a `std::pair<bool, std::string>` with a message to pass to GTEST_SKIP instead. No need for introducing new workaround macros if this could work.

File: test/memspaces/memspace_highest_bandwidth.cpp [hunk 2]
```diff
@@ -9,16 +9,20 @@
 #include "memspace_internal.h"
 #include "test_helpers.h"
 
-static bool canQueryBandwidth(size_t nodeId) {
+static void canQueryBandwidth(size_t nodeId) {
     hwloc_topology_t topology = nullptr;
     int ret = hwloc_topology_init(&topology);
-    UT_ASSERTeq(ret, 0);
+
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_highest_bandwidth.cpp:15]: are these extra empty lines needed? (added in all places in this PR)

File: test/memspaces/memspace_highest_bandwidth.cpp [hunk 3]
```diff
@@ -30,7 +34,10 @@ static bool canQueryBandwidth(size_t nodeId) {
                                   numaNode, &initiator, 0, &value);
 
     hwloc_topology_destroy(topology);
-    return (ret == 0);
+
+    if (ret != 0) {
+        GTEST_SKIP() << "ret is equal to " << ret << ", should be " << 0;
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_highest_bandwidth.cpp:39]: can we make these messages a little more meaningful? this `ret` is bound specifically to the result of `hwloc_memattr_get_value`

File: test/memspaces/memspace_fixtures.hpp [hunk 4]
```diff
@@ -81,10 +82,17 @@ struct memspaceProviderTest : ::memspaceGetTest {
     void SetUp() override {
         ::memspaceGetTest::SetUp();
 
-        if (::memspaceGetTest::IsSkipped()) {
+        if (numa_available() == -1 || numa_all_nodes_ptr == nullptr) {
             GTEST_SKIP();
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_fixtures.hpp:86]: can you add the message (like you did in the other place in this PR)

File: test/memspaces/memspace_helpers.hpp [hunk 5]
```diff
@@ -16,6 +16,8 @@
 #define SIZE_4K (4096UL)
 #define SIZE_4M (SIZE_4K * 1024UL)
 
+#define IS_SKIPPED_OR_FAILED() (HasFatalFailure() || IsSkipped())
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_helpers.hpp:19]: hmm you can change this later (with other PR or something) - but is this helper function only needed in `memspace_helpers.hpp`? no other tests will use it...?

File: test/memspaces/memspace_fixtures.hpp [hunk 6]
```diff
@@ -51,7 +51,7 @@ struct numaNodesTest : ::umf_test::test {
     unsigned long maxNodeId = 0;
 };
 
-using isQuerySupportedFunc = bool (*)(size_t);
+using isQuerySupportedFunc = void (*)(size_t);
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_fixtures.hpp:54]: can you describe in a commit message, in a sentence or two, why this change from void to bool. This seem like the most important part of this PR (it changes very little of UT_ASSERTs, as the title would suggest 😉 )

File: test/memspaces/memspace_highest_bandwidth.cpp (hunk 2 above)
Comment (EuphoricThinking) [reply test/memspaces/memspace_highest_bandwidth.cpp:15]: No, not needed. I have probably added them on "aesthetic" purpose, I'll just delete them.

File: test/memspaces/memspace_fixtures.hpp (hunk 6 above)
Comment (EuphoricThinking) [reply test/memspaces/memspace_fixtures.hpp:54]: Something like the first paragraph of the PR description? Without the sentence with GTEST docs.

File: test/memspaces/memspace_helpers.hpp (hunk 5 above)
Comment (lukaszstolarczuk) [reply test/memspaces/memspace_helpers.hpp:19]: that was just a thought... perhaps `base.hpp` if you wantto put it in hpp header

File: test/memspaces/memspace_fixtures.hpp [hunk 7]
```diff
@@ -65,9 +65,10 @@ struct memspaceGetTest : ::numaNodesTest,
         }
 
         auto [isQuerySupported, memspaceGet] = this->GetParam();
+        isQuerySupported(nodeIds.front());
 
-        if (!isQuerySupported(nodeIds.front())) {
-            GTEST_SKIP();
+        if (IS_SKIPPED_OR_FAILED()) {
+            return;
```
Comment (lukaszstolarczuk) [inline test/memspaces/memspace_fixtures.hpp:71]: hmm... so what happens when we return in the SetUp function? will the test skip or what?

Comment (EuphoricThinking) [reply test/memspaces/memspace_fixtures.hpp:71]: The test has been marked as skipped in isQuerySupported() using GTEST_SKIP(), repeating GTEST_SKIP() would only duplicate the skip message in the output. I'll add a comment in the code

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by PatKamin — APPROVED** [review]
